x-pack/plugins/infra/public/pages/infrastructure/metrics_explorer/use_metric_explorer_state.ts

```diff
@@ -0,0 +1,116 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { useState, useCallback, useContext } from 'react';
+import { StaticIndexPattern } from 'ui/index_patterns';
+import {
+  MetricsExplorerMetric,
+  MetricsExplorerAggregation,
+} from '../../../../server/routes/metrics_explorer/types';
+import { useMetricsExplorerData } from '../../../containers/metrics_explorer/use_metrics_explorer_data';
+import { MetricsExplorerOptionsContainer } from '../../../containers/metrics_explorer/use_metrics_explorer_options';
+import { SourceQuery } from '../../../../common/graphql/types';
+
+export const useMetricsExplorerState = (
```

Comment:
Nice - I like the way you've bundled everything up in to this top level hook composed of others.

Reply:
Yeah... that made testing so much easier.